api/nodes/serializers.py

```diff
@@ -492,7 +492,10 @@ def update(self, instance, validated_data):
                 raise exceptions.NotFound
             if not inst.auth(user):
                 raise exceptions.PermissionDenied
```

Comment:
Now that `add_primary_institution` and `remove_primary_institution` check the user's affiliation, this check should be necessary. Instead, you should handle the possible UserNotAffiliated exception that may be raised.